### PR TITLE
Add presigned url endpoint configuration to AWS

### LINKED ISSUE #8169: Pre-signed URLs: Add Pre-signed URLs endpoint override
Add a configuration property to override the endpoint used to generate pre-signed URLs in lakeFS.
This can be used, for instance, to set an S3 [VPC interface endpoint](https://docs.aws.amazon.com/AmazonS3/latest/userguide/privatelink-interface-endpoints.html#s3-creating-vpc) as the presigned url endpoint instead of the default one, thus allowing greater access control.

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Closes #8169

Added only to AWS since this is where the concept of overriding a presigned URL endpoint is only available there.

### CODE REVIEW

File: docs/security/presigned-url.md [hunk 1]
```diff
@@ -12,7 +12,10 @@ redirect_from:
 
 With lakeFS, you can access data directly from the storage and not through lakeFS using a presigned URL.
 Based on the user's access to an object in the object store, the presigned URL will get read or write access.
-The presign support is enabled for block adapter that supports it (S3, GCP, Azure), and can be disabled by the [configuration]({% link reference/configuration.md %}) (`blockstore.<blockstore_type>.disable_pre_signed`). Note that the UI support is disabled by default.
+The presign support is enabled for block adapter that supports it (AWS, GCP, Azure), and can be disabled by the [configuration]({% link reference/configuration.md %}) (`blockstore.<blockstore_type>.disable_pre_signed`). Note that the UI support is disabled by default.
+
+- It is possible to override the default pre-signed URL endpoint in **AWS** by setting the [configuration]({% link reference/configuration.md %}) (`blockstore.s3.pre_signed_endpoint`).
+This is useful, for example, when you wish to define a [VPC endpoint](https://docs.aws.amazon.com/AmazonS3/latest/userguide/privatelink-interface-endpoints.html#accessing-s3-interface-endpoints) access for the pre-signed URL.
```
Comment (Isan-Rivkin) [inline docs/security/presigned-url.md:18]: Can you please add this also as a possible condition key to the [minimal IAM permission here](https://github.com/treeverse/lakeFS/blob/master/docs/howto/deploy/aws.md?plain=1#L332)

File: pkg/block/s3/adapter.go [hunk 2]
```diff
@@ -120,6 +127,7 @@ func NewAdapter(ctx context.Context, params params.S3, opts ...AdapterOption) (*
 	a := &Adapter{
 		clients:             NewClientCache(cfg, params),
 		preSignedExpiry:     block.DefaultPreSignExpiryDuration,
+		preSignedEndpoint:   params.PreSignedEndpoint,
```
Comment (Isan-Rivkin) [inline pkg/block/s3/adapter.go:130]: Why is this initiated both here and via opts below? Seems redundant in that case.

File: pkg/block/s3/adapter.go [hunk 3]
```diff
@@ -120,6 +127,7 @@ func NewAdapter(ctx context.Context, params params.S3, opts ...AdapterOption) (*
 	a := &Adapter{
```
Comment (Isan-Rivkin) [inline pkg/block/s3/adapter.go:127]: What about `adapter.GetPresignUploadPartURL` endpoint override? I only see it in `adapter.GetPreSignedURL`

File: pkg/block/s3/adapter.go [hunk 4]
```diff
@@ -404,6 +412,11 @@ func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer,
 	presigner := s3.NewPresignClient(client,
 		func(options *s3.PresignOptions) {
 			options.Expires = a.preSignedExpiry
+			if a.preSignedEndpoint != "" {
```
Comment (Isan-Rivkin) [inline pkg/block/s3/adapter.go:415]: Please add some mimimal tests for this feature? 
i.e make make sure we get the presign endpoint when configured, and get regular urls when not etc.

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]
Nice! Left some comments, mostly non blocking besides that question about multipart presign and tests.

File: pkg/block/s3/adapter.go (hunk 4 above)
Comment (Jonathan-Rosenberg) [reply pkg/block/s3/adapter.go]: added

File: pkg/block/blocktest/adapter.go [hunk 5]
```diff
@@ -156,23 +163,41 @@ func testAdapterWalker(t *testing.T, adapter block.Adapter, storageNamespace str
...
+	preSignedURL, exp := getPresignedURLBasicTest(t, adapter, storageNamespace)
+	require.NotNil(t, exp)
+	expectedExpiry := expectedURLExp(adapter)
+	require.Equal(t, expectedExpiry, *exp)
+	_, err := url.Parse(preSignedURL)
+	require.NoError(t, err)
+}
+
+// Test request for a presigned URL with an endpoint override
+func testGetPreSignedURLEndpointOverride(t *testing.T, adapter block.Adapter, storageNamespace string, oe *url.URL) {
```
Comment (Isan-Rivkin) [inline pkg/block/blocktest/adapter.go:175]:
```suggestion
func testGetPreSignedURLEndpointOverride(t *testing.T, adapter block.Adapter, storageNamespace string, overrideEnpoint *url.URL) {
```

**Review by Isan-Rivkin — APPROVED** [review]
Thanks! single non-blocking comment

**Pull request merged**